chassis.go

```diff
@@ -32,66 +33,79 @@ type Chassis struct {
 	VtepLogicalSwitches []string
 }
 
-func (odbi *ovndb) chassisAddImp(name string, hostname string, etype string, ip string) (*OvnCommand, error) {
-	// / Prepare for encap record
-	enCapUUID, err := newRowUUID()
-	if err != nil {
-		return nil, err
+func (odbi *ovndb) chassisAddImp(name string, hostname string, etype []string, ip string,
+	external_ids map[string]string, transport_zones []string, vtep_lswitches []string) (*OvnCommand, error) {
+	if len(name) == 0 {
+		return nil, fmt.Errorf("chassis name cannot be empty")
 	}
-	row := make(OVNRow)
-
-	if len(name) > 0 {
-		row["chassis_name"] = name
+	if len(etype) == 0 {
+		return nil, fmt.Errorf("chassis encap type cannot be empty")
 	}
-
-	if len(etype) > 0 {
-		row["type"] = etype
+	if len(ip) == 0 {
+		return nil, fmt.Errorf("chassis ip cannot be empty")
 	}
-
-	if len(ip) > 0 {
+	//Prepare for encap record
+	var encap_ids [][]string
+	var namedUUID = "named-uuid"
+	var operations []libovsdb.Operation
+	for _, et := range etype {
+		enCapUUID, err := newRowUUID()
+		if err != nil {
+			return nil, err
+		}
+		row := make(OVNRow)
+		row["chassis_name"] = name
+		var encap_id []string
+		encap_id = append(encap_id, namedUUID)
+		encap_id = append(encap_id, enCapUUID)
+		encap_ids = append(encap_ids, encap_id)
 		row["ip"] = ip
-	}
-	if uuid := odbi.getRowUUID(tableEncap, row); len(uuid) > 0 {
-		return nil, ErrorExist
+		row["type"] = et
+		if uuid := odbi.getRowUUID(tableEncap, row); len(uuid) > 0 {
+			return nil, ErrorExist
+		}
+		insertEncapOp := libovsdb.Operation{
+			Op:       opInsert,
+			Table:    tableEncap,
+			Row:      row,
+			UUIDName: enCapUUID,
+		}
+		operations = append(operations, insertEncapOp)
 	}
 
-	insertEncapOp := libovsdb.Operation{
-		Op:       opInsert,
-		Table:    tableEncap,
-		Row:      row,
-		UUIDName: enCapUUID,
-	}
 	// Prepare for chassis record
 	ChassisUUID, err := newRowUUID()
 	if err != nil {
 		return nil, err
 	}
-
 	rowChassis := make(OVNRow)
-
-	var encap_id []string
-	var namedUUID = "named-uuid"
-	encap_id = append(encap_id, namedUUID)
-	encap_id = append(encap_id, enCapUUID)
-
-	rowChassis["encaps"] = encap_id
-	if len(name) > 0 {
-		rowChassis["name"] = name
+	// consolidate encaps
+	var encaps []interface{}
+	encaps = append(encaps, "set" )
+	encaps = append(encaps, encap_ids)
+	rowChassis["encaps"] = encaps
+    rowChassis["name"] = name
+    rowChassis["hostname"] = hostname
```

Comment:
It appears to be alignment problem for the above two rows.

Reply:
It's not showing in the editor. I double checked spaces